### PR TITLE
Rcds

### PR DESCRIPTION
**Pull request by fernandohds564**

### CODE REVIEW

**Review by vellosok75 — APPROVED** [review]

**Review by anacso17 — APPROVED** [review]

File: apsuite/optimization/rcds.py [hunk 1]
```diff
@@ -0,0 +1,457 @@
+"""Robust Conjugate Direction Search Algorithm for Minimization."""
+from threading import Thread as _Thread, Event as _Event
+import logging as _log
+
+import numpy as _np
+import matplotlib.pyplot as _mplt
```
Comment (xresende) [inline apsuite/optimization/rcds.py]: import not being used

File: apsuite/optimization/scanning.py [hunk 2]
```diff
@@ -8,68 +8,19 @@
...
-
-    def calc_obj_fun(self):
-        """Return arrays with dimension of search space."""
-        raise NotImplementedError
 
     def set_limits(self, upper=None, lower=None):
         """."""
         self._upper_limits = upper
         self._lower_limits = lower
         self.ndim = len(upper)
```
Comment (xresende) [inline apsuite/optimization/scanning.py:22]: I think this should be an explicit property attached to private ttribute `self._dim`. The class interface is cleaner this way (no need to look specific implementation code to access all class interface)

Comment (fernandohds564) [reply apsuite/optimization/scanning.py:22]: Thanks! This class is broken, I'm fixing it. This parameter is deprecated with the current standard.

Comment (fernandohds564) [reply apsuite/optimization/scanning.py]: @xresende, could you take a look again, please.

**Review by murilobalves — APPROVED** [review]

**Pull request merged**